pylint/checkers/imports.py

```diff
@@ -490,6 +490,16 @@ def _record_import(self, node, importedmodnode):
                 importedname = node.modname
             else:
                 importedname = node.names[0][0].split('.')[0]
+        if isinstance(node, astroid.ImportFrom) and \
```

Comment:
Alternatively (and better) would be to look for the .level attribute of the node. If it's bigger or equal to 1, than that is a relative import. So "from . import x" should have level 1, while "from .. import z" should have level 2 and so on. The same should happen for "from .c import z".